clients/drcachesim/tests/skip_unit_tests.cpp

```diff
@@ -0,0 +1,151 @@
+        fflush(stderr);                                     \
+        exit(1);                                            \
+    } while (0)
+
+#define CHECK(cond, msg, ...)             \
+    do {                                  \
+        if (!(cond)) {                    \
+            fprintf(stderr, "%s\n", msg); \
+            return false;                 \
+        }                                 \
+    } while (0)
+
+static droption_t<std::string> op_trace_file(DROPTION_SCOPE_FRONTEND, "trace_file", "",
+                                             "[Required] Trace input .zip file",
+                                             "Specifies the input .zip trace file.");
+
+static droption_t<bool> op_verbose(DROPTION_SCOPE_FRONTEND, "verbose", false,
+                                   "Whether to print diagnostics",
+                                   "Whether to print diagnostics");
+
+bool
+test_skip_initial()
+{
+    int view_count = 10;
+    // Our checked-in trace has a chunk size of 20, letting us test cross-chunk
+    // skips.  We verify the chunk size below to ensure updates to that file
+    // remember to set that value.
+    // We check each skip value to ensure the view tool is outputting the
+    // expected instruction count.
+    for (int skip_instrs = 0; skip_instrs < 50; skip_instrs++) {
+        if (op_verbose.get_value())
+            std::cout << "Testing -skip_instrs " << skip_instrs << "\n";
+        // Capture cerr.
+        std::stringstream capture;
+        std::streambuf *prior = std::cerr.rdbuf(capture.rdbuf());
+        // Open the trace.
+        std::unique_ptr<reader_t> iter = std::unique_ptr<reader_t>(
+            new zipfile_file_reader_t(op_trace_file.get_value()));
+        CHECK(!!iter, "failed to open zipfile");
+        CHECK(iter->init(), "failed to initialize reader");
+        std::unique_ptr<reader_t> iter_end =
+            std::unique_ptr<reader_t>(new zipfile_file_reader_t());
+        // Run the tool.
+        std::unique_ptr<analysis_tool_t> tool =
+            std::unique_ptr<analysis_tool_t>(view_tool_create(
+                "", /*thread=*/0, /*skip_refs=*/0, /*sim_refs=*/view_count, "att"));
+        std::string error = tool->initialize_stream(iter.get());
+        CHECK(error.empty(), error.c_str());
+        iter->skip_instructions(skip_instrs);
+        for (; *iter != *iter_end; ++(*iter)) {
+            const memref_t &memref = **iter;
+            CHECK(tool->process_memref(memref), tool->get_error_string().c_str());
+        }
+        // Check the result.
+        std::string res = capture.str();
+        if (op_verbose.get_value())
+            std::cout << "Got: |" << res << "|\n";
+        CHECK(skip_instrs != 0 ||
+                  res.find("chunk instruction count 20") != std::string::npos,
+              "expecting chunk size of 20 in test trace");
+        std::stringstream res_stream(res);
+        std::string line;
+        std::getline(res_stream, line, '\n');
```

Comment:
Would be helpful to add an example of the expected output to a comment here.

Reply:
Done, for each line checked.